cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java

```diff
@@ -133,6 +133,23 @@ public MetricQueryResult query(List<String> metrics, List<String> groupBys, Map<
    * @throws IOException if a network error occurred
    * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
    */
+  public MetricQueryResult query(List<String> metrics, List<String> groupBys, Map<String, String> tags,
```

Comment:
btw, maybe in separate PR, but I recommend have `tags` parameter before `groupBys`, primarily because tags are used much more frequently than groupBys. groupBys are often omitted.